bin/cron_script.sh

```diff
@@ -50,11 +50,11 @@ elif [ "$HOST_CLOUD" = 'azure' ]; then
   sed -i -e '/image_publisher/,+3d' build-azure.json
 
   # Searching and composing URL of to a custom VHD to use, i.e. the latest kubenow stable image
-  vhd_name=$(az storage blob list --account-name "$AZURE_STORAGE_ACCOUNT" --container-name "$AZURE_CONTAINER_NAME" --query [].name --output tsv | grep "kubenow-$CURRENT_VERSION" | grep '.vhd')
+  vhd_name=$(az storage blob list --account-name "$AZURE_STORAGE_ACCOUNT" --container-name "$AZURE_CONTAINER_NAME" --query [].name --output tsv | grep "kubenow-$CURRENT_VERSION[^-abcr]" | grep '.vhd')
   kn_vhd_url="https://kubenow.blob.core.windows.net/system/$vhd_name"
 
   # Finally, inserting image_url attribute in build-azure.json
-  sed -i -e '/os_type/a \         "image_url": "'"$kn_vhd_url"'", ' build-azure.json
+  sed -e "/os_type/a \         \"image_url\": \"$kn_vhd_url\", " build-azure.json
```

Comment:
I revised the sed expression in order to make it easier to expand variable and include double quotes which need to be write in the build json file.